### ISSUE TITLE
Neo 2.7.0 - field data not returned from GraphQL unless query is narrowed

### ISSUE THREAD

**Issue by OscarBarrett** [spicywebau/craft-neo#316]
#### Description

We upgraded to 2.7.0 to test it on one of our sites. GraphQL `entries` queries don't seem to be returning all Neo field data unless the query is narrowed to a single entry. When querying for all entries, about 10% of them return the field data whereas all of them have the field filled out.

For example, where the `addresses` field is a Neo field:

```graphql
{
  entries(section: "section") {
    ... on section_section_Entry {
      slug
      addresses {
        ...on addresses_addressInfo_BlockType {
          __typename
          uid
          address
        }
      }
    }
  }
}
```

```json
{
  "data": {
    "entries": [
      {
        "slug": "my-slug",
        "addresses": []
      },
      ...
```

Narrowed query

```graphql
{
  entries(section: "section", slug: "my-slug") {
    ... on section_section_Entry {
      slug
      addresses {
        ...on addresses_addressInfo_BlockType {
          __typename
          uid
          address
        }
      }
    }
  }
}
```

```json
{
  "data": {
    "entries": [
      {
        "slug": "my-slug",
        "addresses": [
          {
            "__typename": "addresses_addressInfo_BlockType",
            "uid": "dc33454e-a905-41dd-a287-8fbcf24e04eb",
            "address": "123 Main Street, Perth WA 6000"
          }
        ]
      }
    ]
  }
}
```

One temporary fix we found is to add `_count` to the query. The count shows as 0... but it gives us the data.

```graphql
{
  entries(section: "section") {
    ... on section_section_Entry {
      slug
      _count(field: "addresses")
      addresses {
        ...on addresses_addressInfo_BlockType {
          id
          uid
          address
        }
      }
    }
  }
}
```

```json
{
  "data": {
    "entries": [
      {
        "slug": "my-slug",
        "_count": 0,
        "addresses": [
          {
            "id": "85097",
            "uid": "dc33454e-a905-41dd-a287-8fbcf24e04eb",
            "address": "123 Main Street, Perth WA 6000"
          }
        ]
      },
      ...
```

#### Other information
- Neo version: 2.7.0
- Craft version: 3.4.10.1
- (if multi-site) Are the affected Neo fields set to manage blocks on a per-site basis: N/A
- (if multi-site, > 2.4.0) What is the propagation method of the neo field: N/A
- Is eager-loading used? No

**Comment by pvldigital** [comment]
Hi @OscarBarrett,

That is strange. The only thing that was changed recently in regards to GraphQL was the field context. I'll have a look and see what I can find out.

**Comment by pvldigital** [comment]
@OscarBarrett just tested it myself with the follow query: 
```
query ($section: [String]) {
  entries(section: $section) {
    slug
    ... on blog_blog_Entry {
      body {
        __typename
        ... on body_text_BlockType {
          text
        }
      }
    }
  }
}
```

and it seems to be returning the correct data:
```
{
	"data": {
		"entries": [
			{
				"slug": "frasdf",
				"body": [
					{
						"__typename": "body_text_BlockType",
						"text": "sfsadfaf"
					},
					...
					{
						"__typename": "body_text_BlockType",
						"text": null
					}
				]
			},
			...
			{
				"slug": "test",
				"body": [
					{
						"__typename": "body_text_BlockType",
						"text": "asdf"
					}
				]
			},
			{
				"slug": "asdf",
				"body": [
					{
						"__typename": "body_text_BlockType",
						"text": "asdffasddddfda"
					},
					...
				]
			}
		]
	}
}
```

The field you're trying to retrieve is just a plain text field right?

**Comment by OscarBarrett** [comment]
Yes it's just a plain text field. It doesn't seem to happen with a clean DB.

I believe I've found the cause - many neoblocks in our database have a sortOrder of `null`. The new `GROUP BY` clause appears to make only a single row with a sortOrder of `null` returned.

For example, if I include sortOrder in the output of the generated elements query for the addresses fields:

```sql
SELECT `neoblocks`.`sortOrder`, `elements`.`id`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`slug`, `elements_sites`.`siteId`, `elements_sites`.`uri`, `elements_sites`.`enabled` AS `enabledForSite`, `neoblocks`.`fieldId`, `neoblocks`.`ownerId`, `neoblocks`.`typeId`, `neoblocks`.`sortOrder`, `content`.`id` AS `contentId`, `content`.`field_accessibility`, `content`.`field_address`, `content`.`field_align`, `content`.`field_body`, `content`.`field_buttonText`, `content`.`field_buttonType`, `content`.`field_isCircle`, `content`.`field_contentRatio`, `content`.`field_dayOfWeek`, `content`.`field_divider`, `content`.`field_documentGroup`, `content`.`field_documentSection`, `content`.`field_email`, `content`.`field_enableSidebarMenu`, `content`.`field_eventCategory`, `content`.`field_eventDate`, `content`.`field_eventPrice`, `content`.`field_eventTime`, `content`.`field_facebook`, `content`.`field_fax`, `content`.`field_halfRightImage`, `content`.`field_heading`, `content`.`field_hideServiceTimes`, `content`.`field_highlightText`, `content`.`field_instagram`, `content`.`field_isLarge`, `content`.`field_latitude`, `content`.`field_linkedin`, `content`.`field_location`, `content`.`field_longitude`, `content`.`field_mapLink`, `content`.`field_maxColumns`, `content`.`field_metaDescription`, `content`.`field_metaTitle`, `content`.`field_newsCategory`, `content`.`field_office`, `content`.`field_openingHours`, `content`.`field_phone`, `content`.`field_postcode`, `content`.`field_priest`, `content`.`field_requireLogin`, `content`.`field_romanNumerals`, `content`.`field_serviceDate`, `content`.`field_serviceTime`, `content`.`field_serviceTitle`, `content`.`field_services`, `content`.`field_shortDescription`, `content`.`field_isSlider`, `content`.`field_suburb`, `content`.`field_timeItem`, `content`.`field_tintImage`, `content`.`field_twitter`, `content`.`field_website`, `content`.`field_whiteBox`, `content`.`field_youtube`, `structureelements`.`root`, `structureelements`.`lft`, `structureelements`.`rgt`, `structureelements`.`level`, `structureelements`.`structureId`
FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`, `structureelements`.`structureId`
FROM `elements` `elements`
INNER JOIN `neoblocks` `neoblocks` ON `neoblocks`.`id` = `elements`.`id`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
INNER JOIN `content` `content` ON `content`.`elementId` = `elements`.`id`
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `elements`.`id`) AND (EXISTS (SELECT *
FROM `structures`
WHERE (`id` = `structureelements`.`structureId`) AND (`dateDeleted` IS NULL)))
WHERE (`neoblocks`.`fieldId`='93') AND (`elements`.`id` IN ('19345', '19358', '19412', '21296', '22039', '29101', '29272', '29350', '29772', '30599', '31175', '31610', '31950', '32187', '32472', '33469', '33806', '34549', '36055', '36155', '36250', '36315', '36855', '37807', '51954', '53858', '53946', '53959', '54035', '54112', '54200', '54289', '57299', '57411', '57517', '57673', '58282', '58468', '58503', '58560', '58593', '58617', '58644', '58671', '58693', '58719', '58753', '58776', '58799', '58834', '59019', '59110', '59214', '59253', '59296', '59338', '59435', '59618', '59679', '59716', '60045', '60485', '60615', '60679', '60720', '60758', '60813', '60900', '61316', '61519', '61722', '61910', '62970', '63080', '63285', '63345', '63512', '63616', '63799', '63821', '63955', '63990', '64125', '64402', '64463', '64819', '64876', '65139', '66069', '66106', '66186', '66235', '66347', '66421', '66456', '66487', '66547', '66590', '66622', '66765', '66938', '66998', '67076', '67108', '67140', '77677', '78812', '43351', '51620', '53996', '54306', '58016', '58114', '59516', '63630', '64249', '85097', '22507', '27535', '28880', '29884', '57915', '58063', '58874', '58924', '84782', '85100', '32762', '34820', '35493', '23851', '59084', '64702', '25037', '38228', '64547', '44285', '61041')) AND (`elements`.`archived`=FALSE) AND (`elements`.`enabled`=TRUE) AND (`elements`.`dateDeleted` IS NULL) AND (`elements_sites`.`enabled`=TRUE) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)) `subquery`
INNER JOIN `neoblocks` `neoblocks` ON `neoblocks`.`id` = `subquery`.`elementsId`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
INNER JOIN `content` `content` ON `content`.`id` = `subquery`.`contentId`
LEFT JOIN `structureelements` `structureelements` ON (`structureelements`.`elementId` = `subquery`.`elementsId`) AND (`structureelements`.`structureId` = `subquery`.`structureId`)
GROUP BY `neoblocks`.`sortOrder`
ORDER BY `neoblocks`.`sortOrder`
```

![image](https://user-images.githubusercontent.com/3511332/77288001-d72acf00-6d11-11ea-8cc9-aeb5540e9f29.png)

And if I remove the `GROUP BY` clause, I get 136 rows which is the number of entries we have in this section.

**Comment by pvldigital** [comment]
Ah whoops. It needs to be grouped by all selected values instead of just sortOrder. Added a fix and will be releasing it soon.

**Comment by pvldigital** [comment]
Has been released in v2.7.2. Can you try it again after updating?

**Comment by OscarBarrett** [comment]
Confirming it's fixed 👍. Thanks
